assets/model_monitoring/components/src/shared_utilities/df_utils.py

```diff
@@ -240,3 +240,8 @@ def try_get_df_column(df: pyspark_sql.DataFrame, name: str) -> Optional[pyspark_
         return df[name]
     except AnalysisException:
         return None
+
+
+def has_duplicated_columns(df: pyspark_sql.DataFrame) -> bool:
+    """Check if a dataframe has duplicate columns."""
+    return not df.groupBy(df.columns).count().where("count > 1").isEmpty()
```

Comment:
have you tested this? per my test, this will throw below exception if there are duplicated column names in df:
AnalysisException: [AMBIGUOUS_REFERENCE] Reference `<col_name>` is ambiguous, could be: [`<col_name>`, `<col_name>`].
I think we can simply check if df.columns has duplication. like below:
```py
def has_duplicated_columns(df):
  col_names = df.columns
  col_names_set = set(col_names)
  return len(col_names) > len(col_names_set)
```